### PR TITLE
Bug 1917498: use OLM client when installing CRDs

### PR DESCRIPTION
**Pull request by exdx**
**Description of the change:**
Backport of #1888

**Motivation for the change:**
Regression

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage 
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/docs` 
- [ ] Commit messages sensible and descriptive

### CODE REVIEW

File: test/e2e/installplan_e2e_test.go [hunk 1]
```diff
@@ -2715,6 +2721,474 @@ var _ = Describe("Install Plan", func() {
 			Expect(ref.Kind).To(Equal("ConfigMap"))
 		}
 	})
+
+	It("limits installed resources if the scoped serviceaccount has no permissions", func() {
+		c := newKubeClient()
+		crc := newCRClient()
+
+		By("creating a scoped serviceaccount specifified in the operatorgroup")
```
Comment (hasbro17) [inline test/e2e/installplan_e2e_test.go:2729]: Nit: Typo, but it's coming from #1888 so we can ignore it here.
```suggestion
		By("creating a scoped serviceaccount specified in the operatorgroup")
```

**Comment by crawford** [comment]
From the [logs](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/operator-framework_operator-lifecycle-manager/1967/pull-ci-operator-framework-operator-lifecycle-manager-release-4.6-e2e-aws-console-olm/1352313674265530368):

> You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit

Nah. Moving to Quay instead.

/hold

**Comment by crawford** [comment]
Cherry-pick-approved removed. Tests are too flaky and we already have too many fixes to be verified this cycle.

File: test/e2e/installplan_e2e_test.go (hunk 1 above)
Comment (exdx) [reply test/e2e/installplan_e2e_test.go:2729]: Good catch -- will see if I can fix in the 4.5 backport

**Comment by exdx** [comment]
I'm not sure when the docker image issue will be resolved. Can we still move this through? Affected customer was on 4.5 so I'm still trying to get another backport in

/hold cancel
/retest

**Comment by crawford** [comment]
The merge window is now closed, but the next patch manager will take a look later this week.

**Pull request merged**